compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -730,65 +703,6 @@ pub mod coverageinfo {
                 kind: RegionKind::CodeRegion,
             }
         }
-
-        pub fn expansion_region(
-            file_id: u32,
-            expanded_file_id: u32,
-            start_line: u32,
-            start_col: u32,
-            end_line: u32,
-            end_col: u32,
-        ) -> Self {
-            Self {
-                counter: coverage_map::Counter::zero(),
-                file_id,
-                expanded_file_id,
-                start_line,
-                start_col,
-                end_line,
-                end_col,
-                kind: RegionKind::ExpansionRegion,
-            }
-        }
-
-        pub fn skipped_region(
-            file_id: u32,
-            start_line: u32,
-            start_col: u32,
-            end_line: u32,
-            end_col: u32,
-        ) -> Self {
-            Self {
-                counter: coverage_map::Counter::zero(),
-                file_id,
-                expanded_file_id: 0,
-                start_line,
-                start_col,
-                end_line,
-                end_col,
-                kind: RegionKind::SkippedRegion,
-            }
-        }
-
-        pub fn gap_region(
-            counter: coverage_map::Counter,
-            file_id: u32,
-            start_line: u32,
-            start_col: u32,
-            end_line: u32,
-            end_col: u32,
-        ) -> Self {
-            Self {
-                counter,
-                file_id,
-                expanded_file_id: 0,
-                start_line,
-                start_col,
-                end_line,
-                end_col: ((1 as u32) << 31) | end_col,
-                kind: RegionKind::GapRegion,
-            }
-        }
```

Comment:
At this point I thought you should use `git blame` to figure out who added every piece of code you're removing, and ping them, this looks like helpers for some WIP feature.

Reply:
These were added in https://github.com/rust-lang/rust/pull/74733 by @richkadel. Code coverage has since been implemented, so I don't know if these still need to be kept, but Rich should know more.

Reply:
IMO, these methods should not be removed. 

Maybe add a comment instead, to avoid someone wondering about removing them in the future?

These are 3 of 4 methods for setting up the 4 `RegionKind` types. It's true we currently only use `code_region` today, but the coverage mapping format is still actively evolving in LLVM, and we also have some active GitHub issues on Rust coverage support. I have used other `RegionKinds` in the past, and then worked around them when I didn't like the results, but we may use others in the future.

Reply:
Ok, I made these crate-local functions with an `allow(dead_code)` (since `code_region` was only used from within codegen_llvm).